tasks/deploy.json

```diff
@@ -0,0 +1,22 @@
+{
+  "description": "Trigger an r10k deployment.",
+  "puppet_task_version": 1,
+  "parameters": {
+    "environment": {
+      "description": "The environment to deploy.",
+      "type": "String"
```

Comment:
Since there are simple regexes for module and environment names we should probably use them to validate either in the metadata types or the task script itself to prevent shell injections.

Since there are two options here these should probably be optional. 

something like
 Optional[Pattern[/\A[a-z0-9_]+\Z/]]

Reply:
If you're going to do that, I am pretty sure you cannot start or end with an underscore, not sure about leading digits. I don't know where the official restriction description/regex is located, though.